nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -1264,6 +1266,7 @@ in
         (isEnabled "OVERLAY_FS")
       ];
 
+    system.switch.enable = lib.mkIf shouldInstallBootloader true;
```

Comment:
Would it not be better to put this where you turned off `system.switch.enable`?
Since the default value is `true`, It's not clear why this would be needed at all.

Reply:
No, because it could be disabled for any other reason and would still be needed here.

Reply:
I don't think I understood, but if so, at least add a comment. The fact that this is being disabled somewhere for all NixOS tests is an implementation detail of the test infrastructure: qemu-vm is not used exclusively for tests, so this does not look like the right place to me.

Reply:
I feel like I might expect an error instead in that case?

Reply:
Ok, my point was that being true by default in anything but NixOS tests on hydra there's no reason to specify it require explicitely.

Reply:
I understand that, but then we should probably assert here, because otherwise build-disk-image fails with a very non-obvious reason.